### DISCUSSION TITLE
How to convert a *.gif to *.png and KEEP all timestamps?

### DISCUSSION THREAD

**Discussion by pstein** [ImageMagick/ImageMagick#6988] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 10

### Description

How can I convert a source picture source.gif into a target picture target.png (and the corresponding format) WITH KEEPING all timestamps (in particular created,  and last modified)?

target.png should be written into the same directory as source.gif.

**Comment by snibgo** [comment]
See https://imagemagick.org/script/defines.php:
```
preserve-timestamp=true|false 	Preserve file timestamp (mogrify only).
```
For example:
```
magick mogrify -define png:exclude-chunk=date -define preserve-timestamp=true -format png toes.gif
```

**Comment by pstein** [comment]
Thank you.

But your solution works only partially.

The "last modified" timestamp is kept. Good
But the "created" timestamp is set on current/today.
How can I keep "created" timestamp too?

**Reply by pstein** [reply]
...no solution for that?

**Comment by snibgo** [comment]
Please show your input file and the command you used.

**Comment by tobwz** [comment]
Does PNG format support animations (like in GIF) at all?

I think converting an animated *.gif to *.png only converts the first frame.

The only image formats which support animations are APNG and WebP.
Can someone confirm this?

Does magick fully support conversion of an ANIMATION from GIF to APNG/WebP?

**Comment by snibgo** [comment]
IM can read frames from, and write frames to, many animated formats, including GIF, APNG, WEBP, MP4 and MPG. For a complete list, see "`magick -format`" where entries have "+" under "mode".
